src/main/java/PostMoveValidator.java

```diff
@@ -0,0 +1,16 @@
+import java.util.List;
+public class PostMoveValidator {
+    public List<Move> validMoves;
+    /**
+     * Takes in a board instance and a list of semi valid moves
+     * created by MoveGenerator and removes all invalid moves
+     * due to King checks. In more detail, calls MoveTryer for
```

Comment:
Good note. Don't need to include implementation details in javadoc, though, so this should probably become a regular multiline comment.